### PR TITLE
refactor for audio and video constraints fixes #40

### LINKED ISSUE #40: Constraint Settings
Currently, the the options for width/height only affect the rendered canvas, not the video source though. there is no way to set it in the constraints, same goes for other constraints (like echo cancellation, etc...)

### PR DESCRIPTION
**Pull request by leeduan**
Add video and audio constraint settings as props with existing fallback for MediaStreamTrack.getSources using the optional source id

### CODE REVIEW

File: src/react-webcam.js [hunk 1]
```diff
@@ -59,8 +119,8 @@ export default class Webcam extends Component {
 
   componentWillUpdate(nextProps) {
     if (
-      nextProps.videoSource !== this.props.videoSource ||
-      nextProps.audioSource !== this.props.audioSource
+      JSON.stringify(nextProps.audioConstraints) !== JSON.stringify(this.props.audioConstraints) ||
```
[the remark this answers is not in the document]
Comment (brianpmaher) [reply src/react-webcam.js]: Another reason to start using lodash 😉  https://lodash.com/docs/4.17.5#isEqual

**Comment by mozmorris** [comment]
@leeduan thank you, at a glance this looks great and also fixes #40. I'll try to get this reviewed & tested today.

Were you able to test this on Android?

**Comment by leeduan** [comment]
@mozmorris Not tested on android, don't have a device.

**Comment by mozmorris** [comment]
Okay no problem. I have got access to a device, it's just not with me right now.

**Comment by matheusportela** [comment]
Will this PR be merged soon? I'm very much interested in this functionality (in fact, I have my DIY version of it) :)

**Comment by mozmorris** [comment]
@matheusportela done. Thanks @leeduan.

**Comment by leeduan** [comment]
👍

**Pull request merged**